### PR TITLE
Add docs for methods with overloaded signatures

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes #7410

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7406** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7406" target="_blank">Graphite</a> 
    * **PR #7382** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7382" target="_blank">Graphite</a> 
      * **PR #7383** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7383" target="_blank">Graphite</a> 
        * **PR #7410** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7410" target="_blank">Graphite</a>  👈
        * **PR #7384** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7384" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sorbet/sorbet/7410?utm_source=stack-comment).

**Comment by jez** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7406** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7406" target="_blank">Graphite</a> 
    * **PR #7382** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7382" target="_blank">Graphite</a> 
      * **PR #7383** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7383" target="_blank">Graphite</a> 
        * **PR #7410** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7410" target="_blank">Graphite</a>  👈
        * **PR #7384** <a href="https://app.graphite.dev/github/pr/sorbet/sorbet/7384" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sorbet/sorbet/7410?utm_source=stack-comment).

File: website/docs/overloads.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+## Restrictions on overloaded methods
+
+Support for overloaded methods is minimal because there are restrictions on when
+they are allowed to appear and how they are allowed to be used.
+
+Overloaded signatures:
+
+1.  may only appear in RBI files, not in Ruby source files.
+
+1.  prevent the implementation of that method from being type checked.
```
Comment (ilyailya) [inline website/docs/overloads.md:133]: Looks like the ordering in this list is broken

**Review by ilyailya — APPROVED** [review]
Looks good, the only issue I found is the broken ordering in the list

File: website/docs/overloads.md (hunk 1 above)
Comment (jez) [reply website/docs/overloads.md:133]: @ilyailya are you referring to the fact that they all say `1.`? Markdown ordered lists are auto-incrementing when rendered.

Comment (jez) [reply website/docs/overloads.md:133]: You can check the rendered output to confirm this. Using `1.` for all items makes it easy to insert or remove items without needing to update all the numbers

**Pull request merged**